Migrate Xaml.UnitTests from NUnit to xUnit

StephaneDelcroix opened the pull request:
<!-- Please let the below note in for people that find this PR -->
> [!NOTE]
> Are you waiting for the changes in this PR to be merged?
> It would be very helpful if you could [test the resulting artifacts](https://github.com/dotnet/maui/wiki/Testing-PR-Builds) from this PR and let us know in a comment if this change resolves your issue. Thank you!

## Description

This PR migrates the `Controls.Xaml.UnitTests` project from NUnit to xUnit, as part of the effort tracked in #32247.

## Changes

### Test Framework Migration
- Replaced NUnit `[TestFixture]` with xUnit `[Collection]` attributes
- Converted `[Test]` to `[Theory]` with `[MemberData]` for parameterized tests
- Replaced `[Values]` attribute pattern with `XamlInflatorTestData` member data provider
- Updated assertions from NUnit `Assert.That()` / `Assert.AreEqual()` to xUnit `Assert.Equal()` / `Assert.NotNull()` etc.
- Replaced `[SetUp]` / `[TearDown]` with xUnit constructor / `IDisposable.Dispose()` pattern
- Replaced `Assert.Throws<T>` / `Assert.DoesNotThrow` with xUnit equivalents

### Test Organization
- Tests in root folder use `[Collection("Xaml Inflation")]`  
- Tests in Issues/ folder use `[Collection("Issue")]`
- Added collection definitions with `DisableParallelization = true` to maintain test isolation

### Infrastructure
- Added `XamlInflatorTestData` class providing `IEnumerable<object[]>` for inflator variations
- Removed NUnit package references
- Removed dead code (XamlgTests.cs - was entirely commented out)

## Test Results

All **1782 tests passing** (same count as main branch).

Fixes #32247

StephaneDelcroix committed:
Migrate Xaml.UnitTests from NUnit to xUnit

Complete migration of src/Controls/tests/Xaml.UnitTests from NUnit to xUnit testing framework.

rmarinho commented on src/Controls/tests/Xaml.UnitTests/Issues/Gh3082.rt.xaml.cs:
```diff
@@ -37,10 +38,8 @@ public partial class Gh3082 : ContentPage
+				Assert.Single(result.Diagnostics);
 			}
-			else
-				Assert.Ignore($"Test not supported for {inflator}");
```
We lost this one , but I think is fine.

StephaneDelcroix replied:
it is fine

rmarinho commented on src/Controls/tests/Xaml.UnitTests/Issues/Gh7187.rt.xaml.cs:
```diff
@@ -34,10 +35,10 @@ public partial class Gh7187 : ContentPage
 """)
 					.RunMauiSourceGenerator(typeof(Gh7187));
 				//FIXME check diagnostic code
```
Can we check the code now ?

rmarinho reviewed (dismissed):
Did n t saw nothing bad, maybe just some exceptions where we check the lines, I wonder if we can use `XamlExceptionAssert.ThrowsBuildException` , but looks good to merge

StephaneDelcroix committed:
Improve exception tests and remove unnecessary IDisposable

- Bz43694, Bz44216, Issue2450: Use XamlExceptionAssert to check
  exception line/position and message content
- Maui17222: Replace IDisposable with try/finally for cleanup
